tests/src/infra/repos/postgresql/guardian-account.spec.ts

```diff
@@ -11,93 +11,145 @@ const makeSut = (): GuardianAccountRepository => {
 }
 
 describe('GuardianAccountRepository', () => {
+  const input = {
+    firstName: 'any_first_name',
+    lastName: 'any_last_name',
+    email: 'any_email@gmail.com',
+    password: 'any_password',
+    phone: 'any_phone',
+    accessToken: 'any_token',
+    verificationToken: 'any_verification_token'
+  }
+
   describe('AddGuardianRepository', () => {
-    it('Should return a guardian account on success ', async () => {
+    it('Should return null if the email is already registered', async () => {
       const sut = makeSut()
-      const { accessToken, ...guardianData } = makeFakeGuardianData()
-      const { password, verificationTokenCreatedAt, verificationToken, ...guardianDataDb } = guardianData
-
-      const guardian = await sut.add(guardianData)
-
-      expect(guardian).toMatchObject(guardianDataDb)
+      const addSpy = jest.spyOn(sut, 'add')
+      const editedInput = { ...input, email: 'johndoe123@gmail.com' }
+      const firstAttempt = await sut.add(editedInput)
+      const secondAttempt = await sut.add(editedInput)
+      expect(firstAttempt).not.toBeFalsy()
+      expect(secondAttempt).toBeFalsy()
+      expect(addSpy).toBeCalledTimes(2)
+      expect(addSpy).toHaveBeenNthCalledWith(1, editedInput)
+      expect(addSpy).toHaveBeenNthCalledWith(2, editedInput)
     })
 
-    it('Should not return a guardian account if duplicated email or phone is provided', async () => {
+    it('Should return null if the phone is already registered', async () => {
       const sut = makeSut()
-      const firstAttempt = await sut.add(makeFakeGuardianData())
-      const secondAttempt = await sut.add(makeFakeGuardianData())
-      expect(firstAttempt).toBeTruthy()
+      const addSpy = jest.spyOn(sut, 'add')
+      const editedInput = { ...input, phone: '123456789' }
+      const firstAttempt = await sut.add(editedInput)
+      const secondAttempt = await sut.add(editedInput)
+      expect(firstAttempt).not.toBeFalsy()
       expect(secondAttempt).toBeFalsy()
+      expect(addSpy).toBeCalledTimes(2)
+      expect(addSpy).toHaveBeenNthCalledWith(1, editedInput)
+      expect(addSpy).toHaveBeenNthCalledWith(2, editedInput)
     })
-  })
-  describe('LoadAccountByEmailRepository', () => {
-    it('Should return a guardian account on success', async () => {
-      const sut = makeSut()
-      const guardianData = makeFakeGuardianData()
-      await sut.add(guardianData)
-
-      const guardian = await sut.loadByEmail(guardianData.email)
 
-      expect(guardian).toMatchObject(guardianData)
+    it('Should return a guardian account on success ', async () => {
+      const sut = makeSut()
+      const guardian = await sut.add(input) as any
+      expect(guardian).not.toBeFalsy()
+      expect(guardian.id).toBeDefined()
+      expect(guardian).toMatchObject({
+        firstName: input.firstName,
+        lastName: input.lastName,
+        email: input.email,
+        phone: input.phone
+      })
```

Comment:
He was using toMatchObject because the object that comes has some things more that we will not use. Like a "verificationTokenCreatedAt"

Reply:
From what I can see, the `verificationToken` is not being returned by the prism. The `verificationToken: false` on line 42 is redundant, because prism will only return the columns that are `true`. So that can be deleted and the test changed to `toEqual` which will pass